internal/crdReplicator/crdReplicator-operator.go

```diff
@@ -431,68 +427,55 @@ func (c *Controller) StartWatchers() {
 		}
 		for i := range c.RegisteredResources {
 			res := &c.RegisteredResources[i]
-			if !foreigncluster.IsReplicationEnabled(c.getPeeringPhase(remCluster), res) {
+			if !isReplicationEnabled(c.getPeeringPhase(remCluster), res) {
 				continue
 			}
 
-			gvr := convertGVR(res.GroupVersionResource)
 			// if there is not then start one
-			if _, ok := watchers[gvr.String()]; !ok {
+			if _, ok := watchers[res.GroupVersionResource.String()]; !ok {
 				stopCh := make(chan struct{})
-				watchers[gvr.String()] = stopCh
+				watchers[res.GroupVersionResource.String()] = stopCh
 				remoteNamespace, err := c.clusterIDToRemoteNamespace(remCluster)
 				if err != nil {
 					klog.Error(err)
 					continue
 				}
-				go c.watcher(remDynClient, remoteNamespace, gvr, cache.ResourceEventHandlerFuncs{
+				go c.watcher(remDynClient, remoteNamespace, res.GroupVersionResource, cache.ResourceEventHandlerFuncs{
 					AddFunc: func(newObj interface{}) {
 						c.remoteAddWrapper(remDynClient, newObj)
 					},
 					UpdateFunc: func(oldObj, newObj interface{}) {
 						c.remoteModifiedWrapper(remDynClient, newObj)
 					},
 				}, stopCh, c.SetLabelsForRemoteResources)
-				klog.Infof("%s -> starting remote watcher for resource: %s", remCluster, gvr.String())
+				klog.Infof("%s -> starting remote watcher for resource: %s", remCluster, res.GroupVersionResource.String())
 			}
 		}
 		c.RemoteWatchers[remCluster] = watchers
 	}
 	// check if the local watchers are running for each registered resource
 	for _, res := range c.RegisteredResources {
-		gvr := convertGVR(res.GroupVersionResource)
 		// if there is not a running local watcher then start one
-		if _, ok := c.LocalWatchers[gvr.String()]; !ok {
+		if _, ok := c.LocalWatchers[res.GroupVersionResource.String()]; !ok {
 			stopCh := make(chan struct{})
-			c.LocalWatchers[gvr.String()] = stopCh
-			go c.watcher(c.LocalDynClient, metav1.NamespaceAll, gvr, cache.ResourceEventHandlerFuncs{
+			c.LocalWatchers[res.GroupVersionResource.String()] = stopCh
+			go c.watcher(c.LocalDynClient, metav1.NamespaceAll, res.GroupVersionResource, cache.ResourceEventHandlerFuncs{
 				AddFunc:    c.AddFunc,
 				UpdateFunc: c.UpdateFunc,
 				DeleteFunc: c.DeleteFunc,
 			}, stopCh, SetLabelsForLocalResources)
-			klog.Infof("%s -> starting local watcher for resource: %s", c.ClusterID, gvr.String())
+			klog.Infof("%s -> starting local watcher for resource: %s", c.ClusterID, res.GroupVersionResource.String())
 		}
 	}
 }
 
 // StopWatchers stops all the watchers for the resources that have been unregistered.
 func (c *Controller) StopWatchers() {
```

Comment:
Do we have any resource replication process that we need to stop after a given peering phase has been completed? I'm asking this since the operator does not support anymore dynamic registering/unregistering of resources.

Reply:
Yes, we can have to stop some remote watchers when the peering phase changes

Reply:
Hope I have not missed anything, but that logic should still be present (since it was unrelated from resource registration/unregistration, which concerned only the configuration part).